### PR TITLE
TNC Id Module : user input url validation and optimizations

### PR DESCRIPTION
**Pull request by annavane**
## Type of change
- [x ] Bugfix
- [x ] Feature
- [x ] Refactoring (no functional changes, no api changes)

## Description of change

modules/tncIdSystem.js
      - user input URL validations added
      - TNCID recovered from cookie storage if available
      - code optimizations for faster TNCID load
________________________________________
test/spec/modules/tncIdSystem_spec.js
      - added tests for new functions
________________________________________
modules/tncIdSystem.md
      - updated documentation

### CODE REVIEW

File: modules/tncIdSystem.md [hunk 1]
```diff
@@ -1,14 +1,21 @@
-# TNCID UserID Module
+# Overview
 
-### Prebid Configuration
+Module Name: tncIdSystem
+
+# Description
+
+TNCID is a shared persistent identifier that improves user recognition compared to both third-party and first-party cookies. This enhanced identification capability enables publishers and advertisers to consistently recognize their audiences, leading to improved monetization and more precise targeting.  The Newco User ID sub-module adds powerful TNCID user identification technology to your Prebid.js bidstream.
```
Comment (patmmccann) [inline modules/tncIdSystem.md:7]: please don't commit sales copy to the js repo

Comment (annavane) [reply modules/tncIdSystem.md]: [Sales description removed](https://github.com/prebid/Prebid.js/pull/12527/commits/b0976ed114f070bc238ed055e049fca7255f4933)

**Comment by annavane** [comment]
Hi [patmmccann](https://github.com/patmmccann),

Thank you for your feedback so far, let us know if there is anything else we can do to improve our code.

**Comment by annavane** [comment]
Hi [patmmccann](https://github.com/patmmccann),

This module changes are important for our business. Please let us know if there is anything wrong why this pull is still not approved (committed 2 weeks ago).

Thank you!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**